tensorflow/stream_executor/cuda/cuda_gpu_executor.cc

```diff
@@ -1129,6 +1129,37 @@ DeviceDescription *CUDAExecutor::PopulateDeviceDescription() const {
           CU_DEVICE_ATTRIBUTE_MAX_REGISTERS_PER_MULTIPROCESSOR, device_)
           .ValueOrDie());
 
+  // We are loading a dummy ptx kernel to set the device description's
+  // blocks_per_core_limit by calling the CUDA occupancy calculator.  This
+  // value is currently required XLA GPU's CalculateLaunchDimensions()
+  const char* blank_ptx = R"(
+.version 6.0
+.target sm_30
+.address_size 64
+
+        // .globl       testkernel
+.visible .entry testkernel()
+{
+        ret;
+})";
+  const char* kernel_name = "testkernel";
+
+  CUmodule blank_module;
+  CUfunction blank_function;
+  CUDADriver::LoadPtx(context_, blank_ptx, &blank_module);
+  CUDADriver::GetModuleFunction(context_, blank_module, kernel_name,
```

Comment:
Do we need to check for errors on these too?  I have definitely seen cases where we fail to load a module.

Reply:
The CUDADriver wrapped calls already check the result of the base driver calls, so we should be good there

Reply:
AFAICT CUDADriver::LoadPtx returns a bool indicating success/failure.  What is to say that we can safely ignore it?

Reply:
There's also this check which logs an error or the driver's cuModuleLoadDataEx() function:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/cuda/cuda_driver.cc#L622-L623

On top of that, we can check if the returned bool is false, but should we do anything aside from logging the error?

Reply:
I guess I'm less concerned about logging an error (although, seems like we should log a uniform error if we ever return -1), and more concerned that, if LoadPTX returns false, do we expect the `CUmodule blank_module` to be (a) an initialized variable and (b) have a value that's valid to pass to `GetModuleFunction` and the other functions below?

Same for `GetModuleFunction` failing.

Reply:
Ah, I see, makes sense